src/kernel/src/arch/amd64/start.rs

```diff
@@ -83,64 +85,70 @@ fn limine_entry() -> ! {
         x86::controlregs::cr0_write(cr0 & !x86::controlregs::Cr0::CR0_WRITE_PROTECT);
     }
 
-    LIMINE_BOOTINFO.get_response().get().unwrap();
+    LIMINE_BOOTINFO.get_response().unwrap();
+
+    // Set the identity map offset used for fast physical to virtual translations.
+    // The offset is only initialized once at startup so it is safe to write directly.
+    let hhdm_info = LIMINE_HHDM
+        .get_response()
+        .expect("failed to get higher half direct ");
+    unsafe {
+        super::memory::PHYS_MEM_OFFSET = hhdm_info.offset();
+    }
 
     let mut boot_info = LimineBootInfo {
         kernel: unsafe {
             LIMINE_KERNEL
                 .get_response()
-                .get()
                 .expect("no kernel info specified for kernel")
-                .kernel_file
-                .as_ptr()
-                .unwrap()
-                .as_ref()
-                .unwrap()
+                .file()
         },
         maps: alloc::vec![],
         modules: alloc::vec![],
-        rsdp: LIMINE_TABLE.get_response().get().map(|r| {
-            r.address.as_ptr().unwrap() as u64 - 0xffff800000000000
-        } /* TODO: MEGA HACK */),
+        rsdp: LIMINE_TABLE.get_response().map(
+            |r| r.address() as u64 - 0xffff800000000000, /* TODO: MEGA HACK */
```

Comment:
Can this be set to the value we get from the id map, now, I wonder?

Reply:
If that is what that magic number is supposed to be then yeah.